### PR TITLE
Added check to `HandleNpcTalk` to ensure the passed NPC index is within bounds

### PR DESCRIPTION
**Pull request by drunderscore**
NPC index must be `>= -1 && < Main.maxNPCs`.

### CODE REVIEW

**Comment by drunderscore** [comment]
Oops! The client uses `-1` to represent not talking to someone I believe...

**Comment by drunderscore** [comment]
This is good to go now, `-1` is now accepted (and the code was commented about this being a magic value).

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ This is the rolling changelog for TShock for Terraria. Use past tense when addin
   * Do not forget to sign every line you change with your name. (@hakusaro)
   * If there is no section called "Upcoming changes" below this line, please add one with `## Upcoming changes` as the first line, and then a bulleted item directly after with the first change.
 
+## Upcoming changes
```
Comment (hakusaro) [inline CHANGELOG.md]: Can you move this section to the section below it for .12? Everyone added their changes there, which should be here, but it's messy to cleanup in a PR now :/

**Review by hakusaro — DISMISSED** [review]
lgtm_7

lgtm, minor change requested but nothing that blocks merge

**Comment by drunderscore** [comment]
ah... I hope force pushing was okay for you, I'm used to doing it to prevent fix-up commits like that in git history.

**Review by hakusaro — APPROVED** [review]

**Review by Arthri — APPROVED** [review]

**Pull request merged**